### DISCUSSION TITLE
What's the best way to serialize to json a class method?

### DISCUSSION THREAD

**Discussion by johnsturgeon** [nlohmann/json#3943] [Q&A] [answered]
Let's say I have for example this class:
```c++
class Team {
public:
    std::string team_name;
    int rushing_yards;
    int passing_yards;
// rushing + passing yards
    int total_yards();
};
```
How can I represent that as:
```json
{ "team_name": "Panthers", "rushing_yards": 100, "passing_yards": 50, "total_yards": 150 }
```

**Comment by nlohmann** [comment, marked as answer]
See https://json.nlohmann.me/features/arbitrary_types/ - you need to define a `to_json` function in the same namespace of your class. The macros do not work as they would not be able to serialize `total_yards`.

**Reply by johnsturgeon** [reply]
OMG, I got stuck when the compiler told me that const wouldn't work there, but *of course* the method should be marked as const... 

Thanks
